test/cmake/whole_archive/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.24)
```

Comment:
It looks like we still run the CI tests with 3.16.3.

What would happen if you lowered this number to 3.16?  Would the WHOLE_ARCHIVE thing below simply not work?

I guess we need some way to skip this new test when an older version of cmake is found.  Normally we do this via environment variables such as `EMTEST_SKIP_XXX`.  In this case maybe something like `EMTEST_SKIP_NEW_CMAKE`?   And then set that in `.circleci/config.yaml`?

Reply:
> Would the WHOLE_ARCHIVE thing below simply not work?

Yeah I'm afraid it's a [3.24+ only](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#link-features) feature.

> something like EMTEST_SKIP_NEW_CMAKE

If we use an env var to toggle, we need a way to distinguish `whole_archive` from other parameterized tests within the group of `test_cmake`. We can do it by adding a new parameter "cmake_version". But then we may not need the env var at all, as the actual cmake version can be determined by parsing `cmake --version`. Further more, the `cmake_version` may not be necessary as well, as version requirement can be read from `CMakeLists.txt` as well.
I'd like a way that adds least maintenance burden to you.

Reply:
I think you can do:

```
if test_dir == 'whole_archive' and 'EMTEST_SKIP_NEW_CMAKE' in os.environ:
   self.skipTest('EMTEST_SKIP_NEW_CMAKE set')
```

Reply:
The reason we don't do things like `--version` checking is that we don't want tests that implicitly skip based on the environment.  We want to folks (and bots) to have the explicitly ops out of certain tests.